config/filter.d/ufw-port-scan.conf

```diff
@@ -0,0 +1,15 @@
+# fail2ban filter configuration for ufw blocked events (typically port scans)
+#
+# By default every connection attempt blocked by ufw will be
+# logged, e.g.:
+#
+# Sep 14 20:56:41 sierra kernel: [UFW BLOCK] [...] SRC=<REDACTED> DST=<REDACTED> LEN=40 TOS=0x04 PREC=0x00 TTL=48 ID=5614 PROTO=TCP SPT=34092 DPT=23 WINDOW=43012 RES=0x00 SYN URGP=0
+#
+# By carefully setting this filter we ban every IP that tries too many times to
+# connect to non-allowed ports.
+#
+# Author: Michele Bologna https://www.michelebologna.net/
+
+[Definition]
+failregex = .*\[UFW BLOCK\] IN=.* SRC=<HOST>
```

Comment:
Sorry, no way with a catch-alls and unanchored RE.
(see https://github.com/fail2ban/fail2ban/pull/2225#issuecomment-855230817 for example of much safer regex).

Reply:
Hi!

It is very unfortunate that you decided not to catch records like [UFW **LIMIT** BLOCK].
Such records means that the connection rate limit has been exceeded for the port or service.

```
Mar 13 17:46:21 vul-fra1 kernel: [488732.378449] [UFW LIMIT BLOCK] IN=eth0 OUT= MAC=ee:09:8e:00:60:d0:9a:7e:8c:17:ff:f2:08:00 SRC=64.227.158.57 DST=108.61.210.117 LEN=60 TOS=0x00 PREC=0x00 TTL=53 ID=51390 DF PROTO=TCP SPT=60332 DPT=22 WINDOW=64240 RES=0x00 SYN URGP=0 
Mar 13 17:46:26 vul-fra1 kernel: [488737.541033] [UFW BLOCK] IN=eth0 OUT= MAC=ee:09:8e:00:60:d0:9a:7e:8c:17:ff:f2:08:00 SRC=85.217.144.52 DST=108.61.210.117 LEN=40 TOS=0x00 PREC=0x00 TTL=247 ID=54321 PROTO=TCP SPT=52702 DPT=5555 WINDOW=65535 RES=0x00 SYN URGP=0 
Mar 13 17:48:09 vul-fra1 kernel: [488839.976423] [UFW BLOCK] IN=eth0 OUT= MAC=ee:09:8e:00:60:d0:9a:7e:8c:17:ff:f2:08:00 SRC=102.212.241.49 DST=108.61.210.117 LEN=40 TOS=0x08 PREC=0x20 TTL=48 ID=988 PROTO=TCP SPT=15633 DPT=23 WINDOW=6449 RES=0x00 SYN URGP=0  
Mar 13 19:33:42 vul-fra1 kernel: [495172.729122] [UFW LIMIT BLOCK] IN=eth0 OUT= MAC=ee:09:8e:00:60:d0:9a:7e:8c:17:ff:f2:08:00 SRC=139.198.187.234 DST=108.61.210.117 LEN=60 TOS=0x00 PREC=0x00 TTL=47 ID=43515 DF PROTO=TCP SPT=49590 DPT=22 WINDOW=64240 RES=0x00 SYN URGP=0 
```


Many people are protecting SSH and other services using the UFW rate limiting feature.
The simplest example of using this feature (connection attempts to 22/TCP will be limited to 6 within 30 seconds):

```
ufw limit ssh
ufw limit OpenSSH
```

While the rate limiting feature is activated, UFW log starts filling with [UFW LIMIT BLOCK] records.
It's very useful to catch such "well-known port" portscanners on the very early stage to block any further activity from their side.

I'm using a slightly modified RegExp from your post to catch these log records also:

`^\s*\S+ kernel:(?: +\[[^\]]+\])? \[UFW (?:LIMIT )?BLOCK\] (?:\b(?:IN=\w+|OUT=|(?:(?!OUT=|IN=)[A-Z]+=[^ \[]*)+) )*SRC=<ADDR> DST=\S+`

It will be of great benefit to all UFW users if you add this minimal adjustment to your code.
I hope for your positive decision on this request.

See the up-to-date UFW log file could be downloaded from the link below. It contains not so many [LIMIT] records but only because 99.9% portscanners were already blocked for the time I downloaded the log.
https://pastebin.com/TWTnCkD6


BTW: I got the best results in blocking SSH intruders and all other portscanners by enabling bantime.overalljails=true feature, which gave a cumulative blocking effect by combining SSH and UFW-portscan jail databases together.

nano /etc/fail2ban/jail.local
```
[DEFAULT]
bantime.overalljails = true
```